### ISSUE TITLE
Import metadata from CSV

### ISSUE THREAD

**Issue by benwbrum** [benwbrum/fromthepage#1644]
This feature follows on to #1640 and should be built from that branch (unless #1641 has already been merged when the feature is started).

As part of our feature extending the metadata capabilities of FromThePage, we need to allow users to upload spreadsheets of metadata for their works.  This metadata will be stored in in the same `original_metadata` field on the `Work` model as imported IIIF metadata is stored (based on #1640).

## Suggested Implementation
* On the `Settings` screen for a collection add a new button `Upload Metadata` in the right hand column above "Blank Collection".
* When the user hits the button, show them a new form that has
  * A _choose file_ dialogue
  * Explanatory text: "To update metadata for several works within this collection, upload a spreadsheet with one row per work.  See `link` for an example.  (Note that multi-line/description fields cannot be updated.)
  * A link to an example file, which will download a spreadsheet to the user's browser for them to use as an example.  (This file will be provided)
  * A link to an empty file, generated by a new controller action that downloads a CSV to the user's browser containing one row per work in the collection, one column containing the `work.id` and one column containing the `work.title`.
  * A submit button (labeled "Upload")
* When the user uploads a file, we should have a new Rails uploader to accept it.  
* Once the file has been uploaded, processing it should do the following:
  * Create a new CSV object from the uploaded file
  * Create another, empty CSV object to contain errors.  (This file should have headers identical to the uploaded file, except that an additional first column should be called "Error")
  * Parse out the headers.  (These will serve as `label` values in the JSON array.)
  * Loop through rows.  For each row,
    * Identify the corresponding `work` based on the value in the `work_id` column
      * If no work exists with this `work_id`, add the row to the error CSV file with the error, "No work exists with ID #{work_id}"
      * If a work exists with this `work_id`, but it is not in this collection, add the row to the error CSV file with the error, "No work with ID #{work_id} is in collection #{collection.title}"
    * Write or overwrite the corresponding work's `original_metadata` field as follows:
      * Each column value will create a new array element.
      * The `label` for the element will contain the column header
      * The `value` for the element will contain the contents of the cell
  * After all rows are processed, write the error.csv file to the filesystem.
* When processing is finished, display a status screen to the user, containing
  * "Your upload has finished processing. `N` rows were updated successfully, `M` rows encountered errors.  Download the error file here: _link_"
  * The link should download the error file

**Comment by saracarl** [comment]
This feature should not be available/visible for the Individual Researcher plan.

**Comment by benwbrum** [comment]
We have enough data from the end users that we should now be able to continue with real-world data.   (Note that the attached files can be checked into Github as fixtures in the `test_data` directory if wanted.)

Use case:
  * A user uploads the LADI.zip file into a collection, which creates five new works with no metadata.
  * A user uploads the `eaacone...` metadata file to assign metadata.
    * This updates the three `eaacone...` files which have matching filenames.  (I.e. the folder they were imported from in the zip file is mentioned in the `filename` column.
    * This process cannot find the `eaacone...1979` file, so that error appears in the error csv.

Note that some column headers are repeated in the file.  When this happens, we should create an array in the metadata json under the header, containing all values.  So the first row 
[ladi_fixture.zip](https://github.com/benwbrum/fromthepage/files/4453011/ladi_fixture.zip)
(`eaacone...1955..`) should have data matching this ruby hash:
```ruby
{"filename"=>["eaacone_c1955-01"],
 "field_identifier_local"=>
  ["eaacone_c1955-01_p02_0001", "eaacone_c1955-01_p02_0001"],
 "field_extent"=>["1 P\xE1gina"],
 "title"=>
  ["Inaugura\xE7\xE3o de Ponte p\xF5e fim numa antiga reivindica\xE7\xE3o de mais de 50 anos"],
 "field_date_year"=>["1955"],
 "field_date_creation"=>["1955-06-21"],
 "field_date_other"=>[],
 "field_description"=>
  ["Recorte de not\xEDcia do jornal \"Na hora\" a sobre constru\xE7\xE3o da ponte na comunidade de Ivaporanduva"],
 "field_resource_type"=>["Textuais"],
 "field_dcterms_language"=>["Portugu\xEAs"],
 "field_genre"=>["Recortes de not\xEDcias"],
 "rel:aut"=>[],
 "rel:pht"=>[],
 "rel:res"=>[],
 "rel:rpt"=>["Ben\xEA Ferreira"],
 "rel:crp"=>[],
 "rel:jud"=>[],
 "rel:ctb"=>[],
 "field_physical_location"=>["Brasil"],
 "field_subject_topic"=>
  ["Ponte",
   "Reivindica\xE7\xE3o Rio Boc\xF3",
   "Ivaporunduva",
   "Quilombos",
   "Obras"],
 "field_subject_geographic_country"=>["Brasil"],
 "field_subject_geographic"=>
  ["Quilombo de Ivaporanduva, Eldorado, Vale do Ribeira, S\xE3o Paulo, Brasil"],
 "field_foaf_person"=>
  ["Zetinho, Prefeito", "Ribeiro, Tot\xF3", "Leite, Jango"],
 "field_foaf_organization"=>[],
 "field_repository"=>
  ["Equipe de Articula\xE7\xE3o e Assessorias \xE0s Comunidades Negras do Vale do Ribeira"],
 "field_member_of"=>
  ["Cole\xE7\xE3o Digital do Quilombos do Vale do Ribeira SP / PR"],
 "field_condition"=>[],
 "field_notes"=>
  ["Recorte de not\xEDcia do jornal \"Na hora\" a sobre constru\xE7\xE3o da ponte na comunidade de Ivaporanduva",
   "Nome da caixa ou nome da pasta: Recortes de Jornal 1955"]}
```
* Zip file containing folders of images for original upload [ladi_fixture.zip](https://github.com/benwbrum/fromthepage/files/4453013/ladi_fixture.zip)
* Zip file containing ladi_metadata_fixture.zip](https://github.com/benwbrum/fromthepage/files/4453022/ladi_metadata_fixture.zip)
  * CSV file containing metadata for three works 
  * CSV file containing metadata for two more works
[

**Comment by diegoviola** [comment]
@benwbrum I've just implemented all of that in PR #1641, please test and let me know if it needs more changes. Thanks.
